prql-compiler/src/sql/codegen.rs

```diff
@@ -404,62 +403,36 @@ pub(super) fn translate_join(t: &Transform, context: &mut Context) -> Result<Joi
     }
 }
 
-/// Translate a column name. We need to special-case this for BigQuery
-// Ref #852
-fn translate_column(ident: String, context: &Context) -> Vec<sql_ast::Ident> {
-    // TODO: since we have changed how idents are parsed, I think this is no
-    // longer needed; confirm when we renable tests on the semantic branch
-    match context.dialect.dialect() {
-        Dialect::BigQuery => {
-            if let Some((prefix, column)) = ident.rsplit_once('.') {
-                // If there's a table definition, pass it to `translate_ident` to
-                // be surrounded by quotes; without surrounding the table name
-                // with quotes.
-                translate_ident(prefix.to_string(), context)
-                    .into_iter()
-                    .chain(translate_ident(column.to_string(), context))
-                    .collect()
-            } else {
-                translate_ident(ident, context)
-            }
-        }
-        _ => translate_ident(ident, context),
-    }
-}
-
 /// Translate a PRQL Ident to a Vec of SQL Idents.
 // We return a vec of SQL Idents because sqlparser sometimes uses
 // [ObjectName](sql_ast::ObjectName) and sometimes uses
 // [sql_ast::Expr::CompoundIdentifier](sql_ast::Expr::CompoundIdentifier), each of which
 // contains `Vec<Ident>`.
-pub(super) fn translate_ident(ident: String, context: &Context) -> Vec<sql_ast::Ident> {
-    let is_jinja = ident.starts_with("{{") && ident.ends_with("}}");
-
-    if is_jinja {
-        return vec![sql_ast::Ident::new(ident)];
-    }
-
-    match context.dialect.dialect() {
-        // BigQuery has some unusual rules around quoting idents #852 (it includes the
-        // project, which may be a cause). I'm not 100% it's watertight, but we'll see.
-        Dialect::BigQuery => {
-            if ident.split('.').count() > 2 {
-                vec![sql_ast::Ident::with_quote(
-                    context.dialect.ident_quote(),
-                    ident,
-                )]
-            } else {
-                vec![sql_ast::Ident::new(ident)]
-            }
+pub(super) fn translate_ident(
```

Comment:
@aljazerzen thanks for adding things — did I miss something, or is this just resolving merges? Hit me with criticism!

Reply:
translate_ident was operating with `String`, which has concatenated namespace and varname just before. So I allowed namespace and var name into translate_ident separately.